c/xen_common.c

```diff
@@ -329,6 +331,9 @@ set_api_version(xen_session *session)
 
     if (major_version == (int64_t)2)
     {
+        if (minor_version == (int64_t)5)
+            session->api_version = xen_api_version_2_5;
```

Comment:
Bug here. You need an else on the following line, otherwise this case will fall through to xen_api_unknown_version on line 353 and get overwritten.